src/components/AccountInfo.tsx

```diff
@@ -0,0 +1,80 @@
+import Blockies from 'react-blockies';
+import { styled, useTheme } from '@mui/material/styles';
+import { Box, Popover, Typography } from '@mui/material';
+import { useState, useMemo, useCallback, useRef } from 'react';
+import { centerEllipsis, copyToClipboard } from '../utils/strings';
+import { useAppState } from 'src/store';
+
+const AccountIcon = styled(Blockies)`
+  border-radius: 50%;
+`;
+
+export const AccountInfo = () => {
+  const theme = useTheme();
+  const boxRef = useRef();
+  const { account } = useAppState();
+  const [open, setOpen] = useState<boolean>(false);
+  const shortAccount = useMemo(() => centerEllipsis(account || ''), [account]);
+
+  const handleOpen = useCallback(
+    () => {
+      setOpen(true);
+      if (account) {
+        copyToClipboard(account);
+      }
+    },
+    [account]
+  );
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  if (!account) {
+    return null;
+  }
+
+  return (
+    <Box
```

Comment:
There is also a component called `Stack`.

Basically a `Box` , but a focus on display items one-dimensionally. 

You can for example pass the prop `spacing={2}` to define the spacing between items.

Not necessary to refactor to this right now though.